app/Policies/ChapterPolicy.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Policies;
+
+use App\Model\Chapter;
+use App\Model\Course;
+use App\Model\Instructor;
+
+class ChapterPolicy
+{
+    /**
+     * 複数(選択)削除
+     */
+    public function bulkDelete(Instructor $instructor, Chapter $chapter): bool
+    {
+        if ($instructor->isManager()) {
+            // 管理者の場合、配下の講師の講座も削除可能
+            $managerIds = $instructor->managings->pluck('id')->toArray();
+            $managerIds[] = $instructor->id;
+
+            return $chapter->every(function (Chapter $chapter) use ($managerIds) {
+                return in_array($chapter->course->instructor_id, $managerIds, true);
+            });
+        }
+
+        // 講師の場合、自分の講座のみ削除可能
+        return $chapter->every(function (Chapter $chapter) use ($instructor) {
+            return $chapter->course->instructor_id === $instructor->id;
+        });
+    }
```

Comment:
次のように引数にコレクションを渡してしまうというイメージでしたが、いかがでしょう？

```suggestion
    /**
     * 一括削除認可処理
     */
    public function bulkDelete(Instructor $instructor, Collection $chapters): bool
    {
        if ($instructor->isManager()) {
            // 管理者の場合、配下の講師の講座も削除可能
            $managerIds = $instructor->managings->pluck('id')->toArray();
            $managerIds[] = $instructor->id;

            return $chapters->every(function (Chapter $chapter) use ($managerIds) {
                return in_array($chapter->course->instructor_id, $managerIds, true);
            });
        }

        // 講師の場合、自分の講座のみ削除可能
        return $chapters->every(function (Chapter $chapter) use ($instructor) {
            return $chapter->course->instructor_id === $instructor->id;
        });
    }
```

Reply:
大変勉強になりました。
以後、このように処理するようにします。